src/arch/abtd_thread.c

```diff
@@ -106,13 +108,16 @@ static inline void ABTD_thread_terminate(ABTI_thread *p_thread)
         }
     }
 
+    /* Since we use atomic_fetch_or to strictly check if waiting threads exist,
```

Comment:
We don't need to use acquire load for `p_fctx->p_link` (i.e., relaxed is enough), since later we check `p_thread->request` with `fetch_or`. However, it is just confusing. I will use acquire load here.

Note there might be a very tiny performance penalty, while it might result in avoiding a redundant `fetch_or` check.